util/network/src/node_table.rs

```diff
@@ -143,18 +139,30 @@ pub struct Node {
 	pub id: NodeId,
 	pub endpoint: NodeEndpoint,
 	pub peer_type: PeerType,
+	pub attempts: u32,
 	pub failures: u32,
-	pub last_attempted: Option<Tm>,
 }
 
+const DEFAULT_FAILURE_PERCENTAGE: usize = 50;
+
 impl Node {
 	pub fn new(id: NodeId, endpoint: NodeEndpoint) -> Node {
 		Node {
 			id: id,
 			endpoint: endpoint,
 			peer_type: PeerType::Optional,
+			attempts: 0,
 			failures: 0,
-			last_attempted: None,
+		}
+	}
+
+	/// Returns the node's failure percentage (0..100) in buckets of 5%. If there are 0 connection attempts for this
+	/// node the default failure percentage is returned (50%).
+	pub fn failure_percentage(&self) -> usize {
+		if self.attempts == 0 {
+			DEFAULT_FAILURE_PERCENTAGE
+		} else {
+			((self.failures as f64 / self.attempts as f64 * 100.0 / 5.0).round() * 5.0) as usize
```

Comment:
Do we really need to fallback to floats? What about:
```
self.failures * 10_000 / self.attempts
```
(then we get a number within `0 .. 10_000` representing percents with two decimal places)